src/repo_smith/steps/checkout_step.py

```diff
@@ -10,12 +10,19 @@
 class CheckoutStep(Step):
     branch_name: Optional[str]
     commit_hash: Optional[str]
+    start_point: Optional[str]
 
     step_type: StepType = field(init=False, default=StepType.CHECKOUT)
 
     def execute(self, repo: Repo) -> None:
         if self.branch_name is not None:
-            if self.branch_name not in repo.heads:
+            if self.start_point is not None:
+                if self.branch_name in repo.heads:
+                    raise ValueError(
+                        '"start-point" cannot be provided when "branch-name" already exists in checkout step.'
```

Comment:
The error message here is a little misleading. The error isn't that the field is provided, it's because the actual name is provided. I would rather you inject the branch name into the error message and make it clear that it's because it already exists.